test/boost/json_test.cc

```diff
@@ -54,3 +68,163 @@ BOOST_AUTO_TEST_CASE(test_parsing_map_from_null) {
     BOOST_REQUIRE(map1 == map2);
     BOOST_REQUIRE(map1 == empty_map);
 }
+
+namespace {
+
+std::string_view trim(std::string_view sv) {
+    auto it = sv.begin();
+    auto end = sv.end();
+    while (it != end && *it == ' ') {
+        ++it;
+    }
+    return std::string_view(it, end);
+}
+
+std::string_view cmp_and_remove_prefix(std::string_view sv, std::string_view expected) {
+    BOOST_TEST_MESSAGE(format("cmp_and_remove_prefix: {} expected='{}'", sv, expected));
+    trim(sv);
+    BOOST_REQUIRE(sv.starts_with(expected));
+    auto sz = expected.size();
+    sv = sv.substr(sz, sv.size() - sz);
+    return trim(sv);
+}
+
+void verify_parenthesis(std::string_view sv) {
+    static std::unordered_map<char, char> paren_map = {{'{', '}'}, {'[', ']'}, {'(', ')'}, {'<', '>'}};
+
+    char open = sv[0];
+    char close = sv[sv.size() - 1];
+    auto it = paren_map.find(open);
+    if (it == paren_map.end()) {
+        BOOST_FAIL(format("Unexpected delimiters: '{}' '{}'", open, close));
+    }
+    BOOST_REQUIRE_EQUAL(close, it->second);
+}
+
+template <std::ranges::range Range>
+void test_format_range(const char* desc, Range x, std::vector<std::string> expected_strings, std::string expected_delim = ",", bool expect_parenthesis = true) {
+    auto str = seastar::format("{}", x);
+    BOOST_TEST_MESSAGE(format("{}: {}", desc, str));
+
+    auto fmt_str = fmt::format("{}", x);
+    BOOST_REQUIRE_EQUAL(str, fmt_str);
+
+    size_t num_elements = expected_strings.size();
+
+    size_t paren_size = expect_parenthesis ? 2 : 0;
+    size_t min_size = paren_size + (x.empty() ? 0 : (num_elements - 1));
+    BOOST_REQUIRE_GE(str.size(), min_size);
+
+    std::string_view sv = str;
+    if (expect_parenthesis) {
+        verify_parenthesis(sv);
+        sv = sv.substr(1, sv.size() - 2);
+    }
+
+    bool first = true;
+    while (!expected_strings.empty()) {
+        sv = trim(sv);
+        if (!std::exchange(first, false)) {
+            sv = cmp_and_remove_prefix(sv, expected_delim);
+        }
+
+        auto s = *expected_strings.begin();
+        sv = cmp_and_remove_prefix(sv, s);
+        expected_strings.erase(expected_strings.begin());
+    }
+
+    BOOST_REQUIRE(sv.empty());
+}
+
+template <std::ranges::range Range>
+void test_format_range(const char* desc, Range x, std::unordered_set<std::string> expected_strings, std::string expected_delim = ",", bool expect_parenthesis = true) {
+    auto str = seastar::format("{}", x);
```

Comment:
it seems we are testing `seastar::format()` here, which is a very thing wrapper around `fmt::format_to()` . so probably we could have a short note before these tests to explain how/why we need them in scylladb. without more context, a reader might wonder why we would have them here. in other words, it's like testing abseil in scylladb. for instance, how are they related to json related processing?